netket/experimental/driver/tdvp.py

```diff
@@ -39,32 +40,43 @@
 
 
 @dispatch
-def dwdt(state, driver, t, w, *, stage=None):
+def make_odefun(state, driver):
     # pylint: disable=unused-argument
-    raise NotImplementedError(f"dwdt not implemented for {type(state)}")
+    raise NotImplementedError(f"make_odefun not implemented for {type(state)}")
 
 
 @dispatch
-def dwdt(state: MCState, driver, t, w, *, stage: int = None):
-    # pylint: disable=protected-access
-    state.parameters = w
-    state.reset()
+def make_odefun(state: MCState, driver):
+    def odefun(args: dict):
+        # pylint: disable=protected-access
 
-    driver._loss_stats, driver._loss_grad = state.expect_and_grad(
-        driver.generator(t),
-        use_covariance=True,
-    )
-    driver._loss_grad = jax.tree_map(
-        lambda x: driver._loss_grad_factor * x, driver._loss_grad
-    )
+        # unwrap arguments
+        t = args["t"]
+        w = args["w"]
```

Comment:
I really don't like this way of defining the arguments.
Why are we doing that?

Reply:
`host_callback` needs to pass exactly one argument, if I understand it correctly. That can be any PyTree, but `dict` works best with kwargs.